From 6d8445072c9223f721c34031233e0404c283f2a9 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tor=20Arne=20Vestb=C3=B8?= <tor.arne.vestbo@qt.io>
Date: Mon, 12 Feb 2024 11:57:34 +0100
Subject: Enable frameworks by default for shared library builds on Apple
 platforms

Many modern features of the Apple application build and packaging pipeline
require that the libraries are shipped as frameworks, so that they can
embed resources (privacy manifest e.g.), be signed, and have their own
Info.plist.

We build and ship our binary packages already as frameworks, and it has
been the default for release builds for a while. Let's enable it for
debug builds as well, so that developers are testing what we ship
(debug is the default for -developer-build).

The error about debug builds not being compatible with frameworks has
been removed, as this works fine in practice. With CMake we don't add
a '_debug' suffix to the libraries unconditionally for debug builds.

Change-Id: I373b982affd8cf70b215d4a92225467ff1037fe8
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
(cherry picked from commit fee15182949e88682131399c50b679be83a75385)
---
 configure.cmake | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git x/qtbase/configure.cmake y/qtbase/configure.cmake
index 08766d94d0a43b5443bd2549bc3f630c1f6bed55..f84254e0dc72ec269f6b70e81078d5eb8a55a906 100644
--- x/qtbase/configure.cmake
+++ y/qtbase/configure.cmake
@@ -577,7 +577,7 @@ qt_feature("headersclean"
 qt_feature_config("headersclean" QMAKE_PRIVATE_CONFIG)
 qt_feature("framework" PUBLIC
     LABEL "Build Apple Frameworks"
-    CONDITION APPLE AND BUILD_SHARED_LIBS AND NOT CMAKE_BUILD_TYPE STREQUAL Debug
+    CONDITION APPLE AND BUILD_SHARED_LIBS
 )
 qt_feature_definition("framework" "QT_MAC_FRAMEWORK_BUILD")
 qt_feature_config("framework" QMAKE_PUBLIC_QT_CONFIG
@@ -1241,11 +1241,6 @@ qt_configure_add_report_entry(
     MESSAGE "-debug-and-release is only supported on Darwin and Windows platforms.  Qt can be built in release mode with separate debug information, so -debug-and-release is no longer necessary."
     CONDITION INPUT_debug_and_release STREQUAL 'yes' AND NOT APPLE AND NOT WIN32
 )
-qt_configure_add_report_entry(
-    TYPE ERROR
-    MESSAGE "debug-only framework builds are not supported. Configure with -no-framework if you want a pure debug build."
-    CONDITION QT_FEATURE_framework AND QT_FEATURE_debug AND NOT QT_FEATURE_debug_and_release
-)
 qt_configure_add_report_entry(
     TYPE ERROR
     MESSAGE "Static builds don't support RPATH"
